File: nodeVsjavaVsCppVsPython/generalAlgoTest.java
public class generalAlgoTest
{
    static void linearLoopfunc(long x)
    {
        for(long i=1;i<=x;i++)
        {
        }
    }

    static void doubleLoop(long x)
    {
        for(long i=1;i<=x;i++)
        {
            for(long j=1;j<=x;j++)
              {
              }
        }
    }

    static void printTest(long x)
    {
        for(long i=1;i<=x;i++)
        {
            System.out.println("*");
        }
    }

    

    public static void main(String args[])
    {
        long start,finish,timeElapsed;

        start = System.currentTimeMillis();
        linearLoopfunc(1000000000);
        // doubleLoop(100000);
        // printTest(100000);
        finish = System.currentTimeMillis();
        timeElapsed = finish - start;
        System.out.println("10^9 : "+timeElapsed+"ms");

        // start = System.currentTimeMillis();
        // linearLoopfunc(1000000);
        // finish = System.currentTimeMillis();
        // timeElapsed = finish - start;
        // System.out.println("10^6 : "+timeElapsed+"ms");

        // start = System.currentTimeMillis();
        // linearLoopfunc(1000000000);
        // finish = System.currentTimeMillis();
        // timeElapsed = finish - start;
        // System.out.println("10^9 : "+timeElapsed+"ms");

        // start = System.currentTimeMillis();
        // linearLoopfunc(10000000000l);
        // finish = System.currentTimeMillis();
        // timeElapsed = finish - start;
        // System.out.println("10^10 : "+timeElapsed+"ms");

        //10^12 time limit exceeded


    }

}

File: nodeVsjavaVsCppVsPython/generalAlgoTest.js

function linearLoop(l){
    for(i=1;i<=l;i++)
    {
    }
}

function doubleLoop(l){
    for(i=1;i<=l;i++)
    {
        for(j=1;j<=l;j++)
         {
         }
    }
}


function printTest(l){
    for(i=1;i<=l;i++)
    {
        // process.stdout.write("*");
        console.log("*");
    }
}



// console.time('10^3');
//  linearLoop(1000);
// console.timeEnd('10^3');

// console.time('10^6');
//  linearLoop(1000000);
// console.timeEnd('10^6');

// console.time('10^9');
// linearLoop(1000000000);
// console.timeEnd('10^9');

console.time('10^9');
linearLoop(1000000000);
// doubleLoop(100000);
// printTest(100000);
console.timeEnd('10^9');

//time limit exceeded takes a lot of time
// console.time('10^12');
// linearLoop(1000000000000);
// console.timeEnd('10^12');